src/rust/bitbox02-rust-c/src/util.rs

```diff
@@ -147,12 +158,20 @@ pub extern "C" fn rust_util_bytes_mut(buf: *mut c_uchar, len: usize) -> BytesMut
     BytesMut { buf, len }
 }
 
-/// Convert buffer to str
+/// Convert buffer to str.
 ///
-/// * `buf` - Must be a valid pointer to an array of bytes
-/// * `len` - Length of buffer, `buf_ptr[buf_len-1]` must be a valid dereference
+/// * `buf` - Must be a valid pointer to a null terminated array of bytes.
 #[no_mangle]
-pub extern "C" fn rust_util_cstr(buf: *const c_char, len: usize) -> CStr {
+pub extern "C" fn rust_util_cstr(buf: *const c_char) -> CStr {
+    assert!(!buf.is_null());
+    let mut len: usize = 0;
+    let mut b = buf;
+    unsafe {
+        while b.read() != 0 {
```

Comment:
@NickeZ is there any difference from this to `*b != 0`? the read doc says something about "not moving", not sure what it means.

Reply:
I don't think there is any difference. I would probable have written `*b` out of pure habit. Regarding `move` it doesn't really matter as the type you would move in this case is "plan old data" and would therefore simply be copied.

* https://doc.rust-lang.org/std/ops/trait.Deref.html
* https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-dereference-operator

Reply:
Another way of writing similar code is:

```rust
let mut i = 0;
let len = loop {
    if *b == 0 {
        break i;
    }
    i += 1;
}
```

Reply:
Maybe breaking with a value doesn't make sense in this case.. but I wanted you to know about it.

Reply:
very neat trick! thx

Reply:
> Another way of writing similar code is:
> 
> ```rust
> let mut i = 0;
> let len = loop {
>     if *b == 0 {
>         break i;
>     }
>     i += 1;
> }
> ```

you are not advancing `b` anywhere

Reply:
> you are not advancing `b` anywhere

I just wanted to show "break with return" ;) The rest is left as an exercise for the reader!